### ISSUE TITLE
Consider using external lib for data validation

### ISSUE THREAD

**Issue by luispabon** [tomcroft/tantrum_elastic#96]
There seems to be a fair amount of custom validation code in there. Validation has been solved so many times, perhaps we can tap into that?

Might be worth refactoring the validator traits to make use of say https://github.com/Respect/Validation (I really like its fluent syntax) or maybe https://github.com/symfony/Validator.

**Comment by TomChaton** [comment]
Having had a look at the interfaces, I think I prefer Respect.
Do you have the time to do this @luispabon?

**Comment by luispabon** [comment]
Yes. I'll see what I can get done this weekend, I'll also be working on sarah's site either sat or sun :+1:

**Comment by luispabon** [comment]
I'll re-factor to pass current unit tests.

**Comment by TomChaton** [comment]
:muscle: Cheers :smile:
